docs/testplan/build_2tier_network.md

```diff
@@ -0,0 +1,386 @@
+- Protocol Activation: Start the configured protocols in the traffic generator management application. Verify that the traffic generator can send and receive packets through the T0 switch. Conduct initial tests to confirm the proper establishment of connections and protocol functionality.
+
+By following these steps, the traffic generators will be correctly integrated into the network, allowing for accurate traffic simulation, route validation, and performance testing.
+
+## Establish BGP Sessions
+
+### Topology
+
+In this 2-tier network architecture, T0 (Top-of-Rack) switches establish BGP peering with T1 (Leaf) switches to facilitate efficient and scalable routing.
+
+- Many-to-Many Connectivity: Each T0 switch connects to multiple T1 switches via multiple physical links, ensuring load balancing and redundancy.
+- No Direct Connectivity Between Same-Tier Devices: T0 switches do not have direct connections to other T0 switches, and T1 switches do not directly connect to other T1 switches. This enforces a strict hierarchical routing model where all traffic between T0 switches must traverse a T1 switch.
+- BGP Session Distribution: Assume each T1 switch has X logical ports available for BGP peering. Suppose there are Y T0 switches in the network. In this case, each T0 switch establishes X/Y BGP sessions with T1 switches to evenly distribute the session load and optimize routing efficiency.
+
+This topology ensures that traffic is always routed through a T1 switch, providing a structured, predictable, and scalable routing environment.
+
+### Configuration
+
+For example, assume T0-A's Ethernet256 is connected to T1-A's Ethernet0. To establish a BGP session between a T0 switch (T0-A) and a T1 switch (T1-A), the following configurations must be applied.
+
+#### Enable BGP Feature
+
+First, BGP must be enabled on both T0 and T1 switches to allow them to participate in BGP-based route exchange.
+
+T0-A Configuration:
+
+```json
+"FEATURE": {
+    "bgp": {
+        "auto_restart": "enabled",
+        "check_up_status": "false",
+        "delayed": "False",
+        "has_global_scope": "False",
+        "has_per_asic_scope": "True",
+        "high_mem_alert": "disabled",
+        "set_owner": "local",
+        "state": "enabled",
+        "support_syslog_rate_limit": "true"
+    }
+}
+```
+
+T1-A Configuration:
+
+```json
+"FEATURE": {
+    "bgp": {
+        "auto_restart": "enabled",
+        "check_up_status": "false",
+        "delayed": "False",
+        "has_global_scope": "False",
+        "has_per_asic_scope": "True",
+        "high_mem_alert": "disabled",
+        "set_owner": "local",
+        "state": "enabled",
+        "support_syslog_rate_limit": "true"
+    }
+}
```

Comment:
maybe I missed something, but looks like T0 and T1 all uses the same setting. If this is true, it will be better to make this section shorter and keep only the essentials.